paddle/platform/enforce.h

```diff
@@ -162,5 +162,11 @@ inline void throw_on_error(T e) {
     }                                                                   \
   } while (0)
 
+#define PADDLE_ENFORCE_EQ(__VAL0, __VAL1, ...)                              \
```

Comment:
discussed with @Superjom, it's important to use  `PADDLE_ENFORCE_EQ` to print the mismatch value for users to debug if the error exists.